src/scala/com/github/johnynek/bazel_deps/Commands.scala

```diff
@@ -1,94 +1,63 @@
 package com.github.johnynek.bazel_deps
 
-import org.rogach.scallop._
-import org.rogach.scallop.exceptions.ScallopException
+import cats.data.NonEmptyList
+import cats.implicits._
+import com.monovore.decline.{ Command => DCommand, _ }
 import java.io.File
+import java.nio.file.Path
 
-sealed abstract class Command {
-  def validate: Either[String, Command]
-}
+sealed abstract class Command
```

Comment:
I don't know if this `Command` type adds a lot over just calling the side-effecting `FormatDeps` commands directly, especially since you have an `IO` type set up already... having an intermediate type can be handy for testing complicated CLIs, but for lightweight stuff like this it can be a bit boilerplatey.